R/kernel.r

```diff
@@ -242,6 +243,92 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    make_section_header_plain <- function (txt) paste0("# ", txt, ":\n")
+    make_section_header_html <- function (txt) paste0("<h1>", txt, ":</h1>\n")
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Get token under the cursor_pos.
+    # There may be better ways to do that than this.
+    token <- ""
+    for (i in seq(cursor_pos, nchar(code))) {
+        topic_candidate <- utils:::.guessTokenFromLine(code, i)
+        if (!grepl(token, topic_candidate)) break
+        token <- topic_candidate
+    }
+
+    if (nchar(token) == 0) {
+        data <- list()
+        found <- FALSE
+    } else {
+        data <- list()
+
+        class_data <- tryCatch(
+            eval(parse(text = paste0("class(", token, ")"))),
+            error = function (e) character(0))
+        if (length(class_data) != 0) {
+            section_title <- "Class attribute"
+            body <- paste(class_data, collapse = " ")
+            data[["text/plain"]] <- paste0(
+                make_section_header_plain(section_title),
+                body,
+                "\n\n")
+            data[["text/html"]] <- paste0(
+                make_section_header_html(section_title),
+                body,
+                "\n")
+        }
+
+        code_to_get_print_mimebundle <- paste0("IRdisplay::prepare_mimebundle(", token, ")")
+        print_data <- tryCatch(
+            eval(parse(text = code_to_get_print_mimebundle))$data,
+            error = function (e) list())
+        if (length(print_data) != 0) {
+            section_title <- "Printed form"
+            data[["text/plain"]] <- paste0(
+                data[["text/plain"]],
+                make_section_header_plain(section_title),
+                print_data[["text/plain"]],
+                "\n\n")
```

Comment:
This implementation is very simple and easier to extend, but I caught an error when an object has long help document due to the limitation that `sprintf()` can't deal a string longer than 8192 bytes. It seems that using `paste()` is safer from this point of view.
Maybe your point is to avoid repetitions in codes which have a potential risk to cause inconsistency in format, isn't? If so, how about a thing like this?
```R
plain_text_content_builder <- function (content, section_title, new_section) paste0(content, '\n\n#: ', section_title,...)
html_content_builder <- function (...) ...

content_builders <- list(
    'text/plain' = plain_text_content_builder,
    'text/html' = html_content_builder)
```
and
```R
for (mime in names(content_builders)) {
    data[[mime]] <- content_builders[[mime]](data[[mime]], section_title, print_data[[mime]])
}
```